run-tests-parallel.sh

```
@@ -0,0 +1,77 @@
+# This script runs unit tests for all packages in parallel and displays a clean summary.
+# It captures output to temporary files and shows results at the end.
+#
+# Usage:
+#     ./run-tests-parallel.sh
+#
+
+set -euo pipefail
+
+# Create a temporary directory for test outputs
+tmpdir=$(mktemp -d)
+trap "rm -rf $tmpdir" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)